test/parallel/test-cli-node-options-docs.js

```diff
@@ -123,6 +123,20 @@ for (const [, envVar, config] of nodeOptionsCC.matchAll(addOptionRE)) {
   }
 }
 
+{
+
+  const start = /^## Options/m;
+  const end = /^## Environment variables/m;
+  const filteredCLIText = cliText.slice(cliText.search(start), cliText.search(end)).trim();
+  const cliOptionPattern = /^### `(--[a-zA-Z0-9-]+)`/mg;
+  const options = Array.from(filteredCLIText.matchAll(cliOptionPattern)).map((match) => match[1]);
+
+  const sortedOptions = [...options].sort();
+  /* eslint-disable no-restricted-syntax */
+  assert.deepStrictEqual(options, sortedOptions, 'CLI options are not in alphabetical order');
+
```

Comment:
IMO we also want env variables and V8 options to be sorted, it makes little sense to reduce it to our CLI options
```suggestion
  const sections = /^## (.+)$/mg;
  const cliOptionPattern = /^### (?:`-\w.*`, )?`([^`]+)`/mg;
  let match;
  let previousIndex = 0;
  do {
    const sectionTitle = match?.[1];
    match = sections.exec(cliText);
    console.log(sectionTitle, { previousIndex, i: match?.index });
    const filteredCLIText = cliText.slice(previousIndex, match?.index);
    const options = Array.from(filteredCLIText.matchAll(cliOptionPattern), (match) => match[1]);

    assert.deepStrictEqual(options, options.toSorted(), `${sectionTitle} subsections are not in alphabetical order`);
    previousIndex = match?.index;
  } while (match);
```